tests/unit/admin/test-class-wc-rest-payments-settings-controller.php

```diff
@@ -111,14 +118,21 @@ public function set_up() {
 			->disableOriginalConstructor()
 			->getMock();
 
-		$this->mock_wcpay_account = $this->createMock( WC_Payments_Account::class );
-		$this->mock_db_cache      = $this->createMock( Database_Cache::class );
-		$customer_service         = new WC_Payments_Customer_Service( $this->mock_api_client, $this->mock_wcpay_account, $this->mock_db_cache );
-		$token_service            = new WC_Payments_Token_Service( $this->mock_api_client, $customer_service );
-		$order_service            = new WC_Payments_Order_Service( $this->mock_api_client );
-		$action_scheduler_service = new WC_Payments_Action_Scheduler_Service( $this->mock_api_client, $order_service );
-		$mock_rate_limiter        = $this->createMock( Session_Rate_Limiter::class );
-		$mock_dpps                = $this->createMock( Duplicate_Payment_Prevention_Service::class );
+		$this->mock_wcpay_account        = $this->createMock( WC_Payments_Account::class );
+		$this->mock_db_cache             = $this->createMock( Database_Cache::class );
+		$customer_service                = new WC_Payments_Customer_Service( $this->mock_api_client, $this->mock_wcpay_account, $this->mock_db_cache );
+		$token_service                   = new WC_Payments_Token_Service( $this->mock_api_client, $customer_service );
+		$order_service                   = new WC_Payments_Order_Service( $this->mock_api_client );
+		$action_scheduler_service        = new WC_Payments_Action_Scheduler_Service( $this->mock_api_client, $order_service );
+		$mock_rate_limiter               = $this->createMock( Session_Rate_Limiter::class );
+		$mock_dpps                       = $this->createMock( Duplicate_Payment_Prevention_Service::class );
+		$this->mock_localization_service = $this->createMock( WC_Payments_Localization_Service::class );
+
+		$this->mock_localization_service->method( 'get_country_locale_data' )->willReturn(
```

Comment:
I'm struggling to understand what we test here in relation to our changes, can you explain, please?

Reply:
The domestic currency is retrieved from the settings. I need to mock the call to `get_country_locale_data` to return a value, otherwise the test fails because this call will return `null`. 

Since this only affects the tests that go through the `get_settings` flow, I moved the mocking to where it is actually used. I also added a few tests for the retrieval of domestic currency, that were actually missing.

Hope this answers your question and please let me know if there's a better way.

Reply:
Done here https://github.com/Automattic/woocommerce-payments/pull/6920/commits/97edf579f836ba6234c824c3928022670fa19636